src/components/navigation/problem-tab-content.tsx

```diff
@@ -64,18 +86,21 @@ export const ProblemTabContent: React.FC<IProps>
   return (
     <Tabs className={classNames("problem-tabs", context, chatBorder)}
           selectedTabClassName="selected"
-          selectedIndex={ui.activeSectionIndex || 0} //need to add so it won't be undefined
+          selectedIndex={activeIndex || 0}
           data-focus-document={problemPath}
     >
       <div className={classNames("tab-header-row", {"no-sub-tabs": !hasSubTabs})}>
         <TabList className={classNames("tab-list", {"chat-open" : ui.showChatPanel})}>
-          {sections?.map((section) => {
+          {sections?.map((section, index) => {
             const sectionTitle = getSectionTitle(section.type);
             return (
               <Tab
                 className={classNames("prob-tab", context)}
                 key={`section-${section.type}`}
-                onClick={() => handleTabClick(section.type, sectionTitle)}
+                onClick={() => {
+                  handleTabClick(section.type, sectionTitle);
+                  setActiveIndex(index);
```

Comment:
Calling `setActiveIndex` shouldn't be necessary, because the focusDocument should be updated when the tab is changed.  You shouldn't have to worry about how the focusDocument gets updated, but for reference `handleTabClick` calls `updateFocusedDocument` which then triggers the `FocusDocumentTracker` which then figures out the active sub tab and then updates the `focusDocument`.

Reply:
Sent you a message on Slack.